### PR TITLE
gh-92118: fix traceback of exceptions propagated from inside a contextlib.contextmanager

### PR DESCRIPTION
**Pull request by iritkatriel**
#92118

### CODE REVIEW

**Review by hauntsaninja — COMMENTED** [review]
Thanks for the quick fix! I think we might need to do this in the RuntimeError except clause as well.

**Review by gvanrossum — APPROVED** [review]
Looks like a thing.

File: Lib/test/test_contextlib.py [hunk 1]
```diff
@@ -87,6 +88,34 @@ def woohoo():
                 raise ZeroDivisionError()
         self.assertEqual(state, [1, 42, 999])
 
+    def test_contextmanager_traceback(self):
+        @contextmanager
+        def f():
+            try:
+                yield
+            finally:
+                pass
```
Comment (serhiy-storchaka) [inline Lib/test/test_contextlib.py]: Is try/finally really needed?

File: Lib/test/test_contextlib.py [hunk 2]
```diff
@@ -87,6 +88,34 @@ def woohoo():
...
+
+        self.assertEqual(len(frames), 1)
+        self.assertEqual(frames[0].name, 'test_contextmanager_traceback')
+        self.assertEqual(frames[0].line, '1/0')
+
+        try:
+            with f():
+                raise NotImplementedError(42)
+        except NotImplementedError as e:
+            frames = traceback.extract_tb(e.__traceback__)
```
Comment (serhiy-storchaka) [inline Lib/test/test_contextlib.py:111]: What is the difference between this tests and the previous one?

File: Lib/test/test_contextlib.py (hunk 1 above)
Comment (iritkatriel) [reply Lib/test/test_contextlib.py]: It's not needed, I'll remove it.

File: Lib/test/test_contextlib.py (hunk 2 above)
Comment (iritkatriel) [reply Lib/test/test_contextlib.py:111]: The type of the exceptions - RuntimeError goes through a different code path. I'll add a comment.

**Pull request merged**